lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -115,17 +115,21 @@ class CourseList(CourseViewMixin, ListAPIView):
     def get_queryset(self):
         course_ids = self.request.QUERY_PARAMS.get('course_id', None)
 
-        course_descriptors = []
+        results = []
         if course_ids:
             course_ids = course_ids.split(',')
             for course_id in course_ids:
                 course_key = CourseKey.from_string(course_id)
                 course_descriptor = courses.get_course(course_key)
-                course_descriptors.append(course_descriptor)
+                results.append(course_descriptor)
         else:
-            course_descriptors = modulestore().get_courses()
+            results = modulestore().get_courses()
+
+        # Ensure only course descriptors are returned.
+        results = [course for course in results if course.scope_ids.block_type == 'course']
```

Comment:
Nit: If you use `()` instead of `[]` in this line and the next line, then you'll only put 1 list in memory, instead of 2.